feature_engine/selection/recursive_feature_addition.py

```diff
@@ -0,0 +1,269 @@
+        after fitting.
+
+    scoring: str, default='roc_auc'
+        Desired metric to optimise the performance of the estimator. Comes from
+        sklearn.metrics. See the model evaluation documentation for more options:
+        https://scikit-learn.org/stable/modules/model_evaluation.html
+
+    threshold: float, int, default = 0.01
+        The value that defines if a feature will be kept or removed. Note that for
+        metrics like roc-auc, r2_score and accuracy, the thresholds will be floats
+        between 0 and 1. For metrics like the mean_square_error and the
+        root_mean_square_error the threshold will be a big number.
+        The threshold must be defined by the user. Bigger thresholds will retain less
+        features.
+
+    cv : int, default=3
+        Cross-validation fold to be used to fit the estimator.
+
+    Attributes
+    ----------
+
+    initial_model_performance_: float
+        performance of the model built using the original dataset with all the features.
+
+    feature_importances_: pandas series
+        A pandas Series containing the feature names in the axis, and the performance
+        derived from the model trained on the entire dataset, as values. The Series is
+        ordered from least important to most important feature.
+
+    performance_drifts_: dict
+        A dictionary containing the feature, and the change in performance incurred
+        when training a model without that feature.
+
+
+    Methods
+    -------
+
+    fit: finds important features
+
+    transform: removes non-important / non-selected features
+
+    fit_transform: finds and removes non-important features
+
+    """
+
+    def __init__(
+        self,
+        estimator=RandomForestClassifier(),
+        scoring: str = "roc_auc",
+        cv: int = 3,
+        threshold: Union[int, float] = 0.01,
+        variables: Variables = None,
+    ):
+
+        if not isinstance(cv, int) or cv < 1:
+            raise ValueError("cv can only take positive integers bigger than 1")
+
+        if not isinstance(threshold, (int, float)):
+            raise ValueError("threshold can only be integer or float")
+
+        self.variables = _check_input_parameter_variables(variables)
+        self.estimator = estimator
+        self.scoring = scoring
+        self.threshold = threshold
+        self.cv = cv
+
+    def fit(self, X: pd.DataFrame, y: pd.Series):
+        """
+
+        Args
+        ----
+
+        X: pandas dataframe of shape = [n_samples, n_features]
+           The input dataframe
+
+        y: array-like of shape (n_samples)
+           Target variable. Required to train the estimator.
+
+
+        Returns
+        -------
+
+        self
+        """
+
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # find numerical variables or check variables entered by user
+        self.variables = _find_or_check_numerical_variables(X, self.variables)
+
```

Comment:
Should we check for NA?  the model will fail if there are NA in the df. We can catch it here, or let sklearn send the error. Whad did we do for the recursive feature elimination?